pkg/storage/s3/s3.go

```diff
@@ -129,43 +210,74 @@ func (d *driver) createAndTagBucket(svc *s3.S3, installConfig *installer.Install
 	encryptionConfig := &s3.ServerSideEncryptionConfiguration{Rules: encryptionRules}
 	bucketEncryptionInput := &s3.PutBucketEncryptionInput{Bucket: aws.String(d.Config.Bucket), ServerSideEncryptionConfiguration: encryptionConfig}
 
-	_, err := svc.PutBucketEncryption(bucketEncryptionInput)
+	_, err = svc.PutBucketEncryption(bucketEncryptionInput)
 	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			util.UpdateCondition(cr, opapi.StorageExists, operatorapi.ConditionFalse, aerr.Code(), aerr.Error())
+		}
+
 		return err
 	}
 
-	customResourceStatus.Storage.Managed = true
+	cr.Status.Storage.Managed = true
+
+	util.UpdateCondition(cr, opapi.StorageExists, operatorapi.ConditionTrue, "S3 Bucket Exists", "")
 
 	return nil
 }
 
-func (d *driver) createOrUpdatePrivateConfiguration(accessKey string, secretKey string) error {
-	data := make(map[string]string)
+// RemoveStorage deletes the storage medium that we created
+func (d *driver) RemoveStorage(cr *opapi.ImageRegistry) error {
+	if !cr.Status.Storage.Managed {
+		return fmt.Errorf("the S3 bucket is not managed by the image registry operator, so we can't delete it.")
+	}
 
-	data["REGISTRY_STORAGE_S3_ACCESSKEY"] = accessKey
-	data["REGISTRY_STORAGE_S3_SECRETKEY"] = secretKey
+	svc, err := d.getSVC()
+	if err != nil {
+		return err
+	}
+	_, err = svc.DeleteBucket(&s3.DeleteBucketInput{
+		Bucket: aws.String(d.Config.Bucket),
+	})
+
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			util.UpdateCondition(cr, opapi.StorageExists, operatorapi.ConditionTrue, aerr.Code(), aerr.Error())
```

Comment:
it's probably condition unknown at this point... we don't know if the bucket got deleted or not, given that we got an error.